File: Solved_ac_CLASS2/Main_2609_최대공약수와_최소공배수.java
import java.io.*;
import java.util.*;

public class Main_2609_최대공약수와_최소공배수 {
	public static void main(String[] args) throws Exception {
		System.setIn(new FileInputStream("res/input_bj_2609.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int num1 = Integer.parseInt(st.nextToken());
		int num2  = Integer.parseInt(st.nextToken());
		
		int min = Math.min(num1, num2);
		int common_divisor;
		
		// 최대 공약수
		for(common_divisor = min; common_divisor>=1; common_divisor--) {
			if(num1 % common_divisor == 0 && num2 % common_divisor == 0) {
				System.out.println(common_divisor);
				break;
			}
		}
				
		int i=2;
		int common_multiple = num1;
		
		// 최소 공배수
		for(;;) {			
			if(common_multiple % num2 == 0) {
				System.out.println(common_multiple);
				break;
			}
			
			common_multiple = num1 * i;
			i++;
		}

	}
}

File: 정수론 & 조합론 부수기/Main 2609 최대공약수와 최소공배수.kt
package `정수론 & 조합론 부수기`

import java.util.*
import java.io.*

fun main() {
    val path = "C:\\Users\\Samsung\\Desktop\\코틀린 알고리즘\\src\\main\\kotlin\\정수론 & 조합론 부수기\\res\\2609.txt"
    val br = BufferedReader(File(path).bufferedReader())

    val st = StringTokenizer(br.readLine())
    val N = st.nextToken().toLong()
    val M = st.nextToken().toLong()

    val gcd = GCD(N, M)
    val lcm = LCM(N, M, gcd)
    println(gcd)
    println(lcm)
} // End of main

private fun GCD(a: Long, b: Long): Long {
    if (a % b == 0L) return b
    return GCD(b, a % b)
} // End of GCD

private fun LCM(N: Long, M: Long, gcd: Long): Long {
    return (N * M) / gcd
} // End of LCM